src/firebolt/async_db/cursor.py

```diff
@@ -327,25 +337,37 @@ async def _api_request(
             content=query,
         )
 
+    async def _async_api_request(self, query: str) -> Response:
+        """Do query request using SET async_execution=1."""
+        return await self._client.request(
+            url="/",
+            method="POST",
+            params={
+                "database": self.connection.database,
+                "output_format": JSON_OUTPUT_FORMAT,
+                **self._set_parameters,
+                "async_execution": 1,
+                "advanced_mode": 1,
+            },
+            content=query,
+        )
+
     async def _do_execute(
         self,
         raw_query: str,
         parameters: Sequence[Sequence[ParameterType]],
-        set_parameters: Optional[Dict] = None,
         skip_parsing: bool = False,
+        async_execution: bool = False,
     ) -> None:
         self._reset()
-        if set_parameters is not None:
-            logger.warning(
-                "Passing set parameters as an argument is deprecated. Please run "
-                "a query 'SET <param> = <value>'"
-            )
+        if async_execution and "use_standard_sql" in self._set_parameters:
+            raise AsyncExecutionUnavailableError()
```

Comment:
let's add a test for this as well

Reply:
I had planned to do this in a future PR, but I can do it here if you'd prefer.

Reply:
It's generally a good idea to add tests along with the modified code so I'd prefer to have it here.

Reply:
This is addressed above.